src/index.js

```diff
@@ -280,7 +281,7 @@ export default class GoCommerce {
       }));
     } else {
       return Promise.reject(
-        "Invalid paymentDetails - must have an order_id, an amount and a stripe_token or a paypal_payment_id and paypal_user_id"
+        "Invalid paymentDetails - must have an order_id, an amount, a provider, and a stripe_token or a paypal_payment_id and paypal_user_id"
```

Comment:
I was thinking just doing the same validation that you do in the server. Based on the provider you check that the right values are set. 

Also, I am comfortable with just delegating that down to the server.